app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -211,64 +222,61 @@ private function parseReferralCode($referralCode)
-        ];
+        if (!is_null($user)) {
+            return $user->id;
+        }
 
-        foreach ($userFieldsToLookFor as $field => $value)
-        {
-            if ($value) {
-                info('getting user with the '.$field.' field', [$field => $value]);
-                $user = gateway('northstar')->asClient()->getUser($field, $value);
-            }
+        $userData = [
+            'email' => $recordEmail,
+            'mobile' => $recordMobile,
+            'first_name' => $record['First name'],
+            'last_name' => $record['Last name'],
+            'addr_street1' => $record['Home address'],
+            'addr_street2' => $record['Home unit'],
+            'addr_city' => $record['Home city'],
+            'addr_state' => $record['Home state'],
+            'addr_zip' => $record['Home zip code'],
+            'source' => env('NORTHSTAR_CLIENT_ID'),
+        ];
 
-            if ($user) {
-                break;
-            }
+        $recordEmailOptIn = $record['Opt-in to Partner email?'];
+        if (isset($recordEmailOptIn)) {
+            $userData['email_subscription_status'] = str_to_boolean($recordEmailOptIn);
         }
 
-        if (is_null($user)) {
-            $userData = [
-                'email' => $record['Email address'],
-                'mobile' => $record['Phone'],
-                'first_name' => $record['First name'],
-                'last_name' => $record['Last name'],
-                'addr_street1' => $record['Home address'],
-                'addr_street2' => $record['Home unit'],
-                'addr_city' => $record['Home city'],
-                'addr_state' => $record['Home state'],
-                'addr_zip' => $record['Home zip code'],
-                'source' => env('NORTHSTAR_CLIENT_ID'),
-            ];
-
-            if ($record['Phone']) {
-                $userData['sms_status'] = $this->transformSmsStatus($record['Opt-in to Partner SMS/robocall']);
-            }
+        // Note: Not a typo -- this column name does not have the trailing question mark.
+        $recordSmsOptIn = $record['Opt-in to Partner SMS/robocall'];
+        if (isset($recordSmsOptIn) && isset($recordMobile) & !empty($recordMobile)) {
+            $userData['sms_status'] = str_to_boolean($recordSmsOptIn) ? 'active' : 'stop';
+        }
 
-            $user = gateway('northstar')->asClient()->createUser($userData);
+        $user = gateway('northstar')->asClient()->createUser($userData);
 
-            if ($user->id) {
-                info('created user', ['user' => $user->id]);
-            } else {
-                throw new Exception(500, 'Unable to create user: ' . $record['Email address']);
-            }
+        if ($user->id) {
+            info('created user', ['user' => $user->id]);
+        } else {
+            throw new Exception(500, 'Unable to create user: ' . $recordEmail);
         }
 
-        return $user;
+        return $user->id;
```

Comment:
Is it more helpful here to return the full `NorthstarUser` object in case we wanted to read an existing user's profile fields for something? (I also tend to err on the side of passing & returning objects whenever possible since you can add type-checking then too… ensuring you pass a `User` with a _user_ ID, rather than just any string ID).